c++/CMakeLists.txt

```diff
@@ -25,30 +25,15 @@ set(INSTALL_TARGETS_DEFAULT_ARGS
 
 # Options ======================================================================
 
-option(BUILD_TOOLS "Build command-line tools and compiler." ON)
 option(BUILD_TESTING "Build unit tests and enable CTest 'check' target." ON)
 option(EXTERNAL_CAPNP "Use the system capnp binary, or the one specified in $CAPNP, instead of using the compiled one." OFF)
-option(CAPNP_LITE "Compile Cap'n Proto in 'lite mode', in which all reflection APIs (schema.h, dynamic.h, etc.) are not included. Produces a smaller library at the cost of features. All programs built against the library must be compiled with -DCAPNP_LITE. Requires EXTERNAL_CAPNP to build the tests." OFF)
+option(CAPNP_LITE "Compile Cap'n Proto in 'lite mode', in which all reflection APIs (schema.h, dynamic.h, etc.) are not included. Produces a smaller library at the cost of features. All programs built against the library must be compiled with -DCAPNP_LITE. Requires EXTERNAL_CAPNP." OFF)
 
 # Check for invalid combinations of build options
-if(NOT BUILD_TOOLS AND BUILD_TESTING AND NOT EXTERNAL_CAPNP)
-  # Not *all* of the tests require the capnp compiler, and those that do could be excluded
-  # when not building the tools, but it's easier to just have a blanket rule.
-  message(SEND_ERROR "Tests (BUILD_TESTING) cannot be build without either BUILD_TOOLS or EXTERNAL_CAPNP.")
+if(CAPNP_LITE AND NOT EXTERNAL_CAPNP)
+  message(SEND_ERROR "You must set EXTERNAL_CAPNP when using CAPNP_LITE.")
```

Comment:
I think this should be `if(CAPNP_LITE AND BUILD_TESTING AND NOT EXTERNAL_CAPNP)`, otherwise it's too restrictive: the `CAPNP_LITE` library is prevented from being built as a standalone target.